util/bigint/src/uint.rs

```diff
@@ -685,6 +687,17 @@ macro_rules! construct_uint {
 			}
 
 			#[inline]
+			fn to_hex(&self) -> String {
```

Comment:
Doing this via `LowerHex` is indeed more idiomatic and should save allocation costs while formatting it as part of any other message (this is actually something to be mildly concerned about -- @arkpar and I found out recently that enabling any logging will cause the format arguments for all logs at that level, even with differing targets, to be evaluated)